src/compiler/checker.ts

```diff
@@ -2895,6 +2902,34 @@ namespace ts {
             return unknownType;
         }
 
+        function getTypeOfBasePropertyDeclaration(declaration: VariableLikeDeclaration) {
+            if (declaration.parent.kind === SyntaxKind.ClassDeclaration) {
+                const property = getPropertyFromBaseInterfaces(<ClassLikeDeclaration>declaration.parent, declaration.symbol.name);
+                if (property) {
+                    return getTypeOfSymbol(property);
+                }
+            }
+        }
+
+        function getPropertyFromBaseInterfaces(declaration: ClassLikeDeclaration, propertyName: string): Symbol {

```

Comment:
Why only base interfaces? Shouldn't we look at the base class for a contextual type?